pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/observables/ObservablesTNCudaOperator.hpp

```diff
@@ -0,0 +1,295 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc. and contributors.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cutensornet.h>
+#include <tuple>
+#include <vector>
+
+#include "ObservablesTNCuda.hpp"
+#include "TensorCuda.hpp"
+#include "Util.hpp"
+#include "cuGates_host.hpp"
+#include "tncudaError.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::Util;
+using namespace Pennylane::LightningTensor;
+using namespace Pennylane::LightningTensor::TNCuda;
+
+template <class T> using vector1D = std::vector<T>;
+
+template <class T> using vector2D = std::vector<vector1D<T>>;
+
+template <class T> using vector3D = std::vector<vector2D<T>>;
+} // namespace
+/// @endcond
+
+namespace Pennylane::LightningTensor::TNCuda::Observables {
+
+// Current design allows multiple measurements to be performed for a
+// given circuit.
+/**
+ * @brief ObservableTNCudaOperator Class.
+ *
+ * This class creates `custatenetTensorNetwork` Operator from
+ * `ObservablesTNCuda` objects for measurement purpose. Since the NamedObs,
+ * HermitianObs, TensorProdObs and Hamiltionian objects can be encapsulated in a
+ * `cutensornetNetworkOperator_t` instance, only one ObservableTNCudaOperator
+ * class is designed here. Note that a `cutensornetNetworkOperator_t` object can
+ * only be created and destroyed by creating a new ObservableTNCudaOperator
+ * object, which ensures its lifetime is aligned with that of data associated to
+ * it.
+ *
+ * @tparam StateTensorT State tensor class.
+ */
+template <class StateTensorT> class ObservableTNCudaOperator {
+  public:
+    using PrecisionT = typename StateTensorT::PrecisionT;
+    using CFP_t = typename StateTensorT::CFP_t;
+    using ComplexT = typename StateTensorT::ComplexT;
+    using obs_key =
+        std::tuple<std::string, std::vector<PrecisionT>, std::size_t>;
+
+  private:
+    cutensornetNetworkOperator_t obsOperator_{
+        nullptr}; // cutensornetNetworkOperator operator
+
+    const StateTensorT &state_tensor_; // Quantum state to be measured
+
+    const std::size_t numObsTerms_;    // Number of observable terms
```

Comment:
Nitpick, that or cap comments below.
```suggestion
    const StateTensorT &state_tensor_; // quantum state to be measured

    const std::size_t numObsTerms_;    // number of observable terms
```